internal/node/agentcomms.go

```diff
@@ -14,93 +14,87 @@ import (
 
 // Called when the node server gets a log entry via internal NATS. Used to
 // package and re-mit with additional metadata on $NEX.logs...
-func handleAgentLog(mgr *MachineManager) func(m *nats.Msg) {
-	return func(m *nats.Msg) {
-		tokens := strings.Split(m.Subject, ".")
-		vmId := tokens[1]
-
-		vm, ok := mgr.allVms[vmId]
-		if !ok {
-			mgr.log.Warn("Received a log from a VM we don't know about. Rejecting")
-			return
-		}
-
-		var logentry agentapi.LogEntry
-		err := json.Unmarshal(m.Data, &logentry)
-		if err != nil {
-			mgr.log.Error("Failed to unmarshal log entry from agent", slog.Any("err", err))
-			return
-		}
-
-		outLog := emittedLog{
-			Text:      logentry.Text,
-			Level:     slog.Level(logentry.Level),
-			MachineId: vmId,
-		}
-
-		bytes, err := json.Marshal(outLog)
-		if err != nil {
-			mgr.log.Error("Failed to marshal our own log entry", slog.Any("err", err))
-			return
-		}
-
-		_ = mgr.nc.Publish(logPublishSubject(vm.namespace, mgr.publicKey, vm.workloadSpecification.DecodedClaims.Subject, vmId), bytes)
+func (mgr *MachineManager) handleAgentLog(m *nats.Msg) {
+	tokens := strings.Split(m.Subject, ".")
+	vmId := tokens[1]
```

Comment:
do we need a length check here?  if tokens == 0, this will panic

Reply:
since we're the ones who chose the subscription subject, we won't get a message unless it came in on our sub topic, which should have at least 1 token

Reply:
That said, your logic stands.